### PR TITLE
feat: rotate zero width sider trigger in right mode

### PR DESCRIPTION
**Pull request by kagawagao**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/18026
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     rotate zero width sider trigger in right mode      |
| 🇨🇳 Chinese |    当 `Sider` 在右边时，翻转 `trigger` 方向      |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
How about `border-radius`?

**Comment by kagawagao** [comment]
@afc163 updated

File: components/layout/style/index.less [hunk 1]
```diff
@@ -114,6 +114,7 @@
 
         &-right {
           left: -@layout-zero-trigger-width;
+          border-radius: @border-radius-base 0 0 @border-radius-base;
```
Comment (afc163) [inline components/layout/style/index.less]: no rotate then?

Comment (kagawagao) [reply components/layout/style/index.less]: use one of them can work well as expected
difference is `rotate` will make menu icon inverted, but `border-radius` will not

Comment (kagawagao) [reply components/layout/style/index.less]: https://codesandbox.io/s/antd-reproduction-template-d34hf

Comment (afc163) [reply components/layout/style/index.less]: 👌

**Review by afc163 — APPROVED** [review]

**Comment by kagawagao** [comment]
close #18026

**Pull request merged**